TempleEight/docs/guide/authentication.md

```diff
@@ -3,3 +3,186 @@ id: authentication
+{"id":"74045036-87ed-11ea-9edc-0242c0a81003","foo":"hello","bar":123}
+
+```
+Now we included a valid token in the Authorization header, the request was authorized and the entity successfully created.
+
+## Services with `#auth` Metadata
+You may have noticed that so far we have ignored the `ExampleUser` service we previously defined. 
+That's because it acts as any other service, so any example given in this guide could have been replace with a call to that service too.
+
+However, it has one distinguishing feature: a single token is only able to create a single entity in this service. 
+This means means that there is a one-to-one mapping between the `Auth` service, and an entity in the `ExampleUser` service.
+This makes the `ExampleUser` service perfect for storing additional metadata about a user, such as their name, address or anything that fits your business needs.
+
+## How Authentication is Implemented
+Behind the scenes, lots of additional code is executed when authentication is added to your project, to protect your data and your users.
+To best explain this let's consider how a new user registers with your service:
+
+#### Pre-request Preparation
+Before any request is even issued to your `Auth` service, several steps are taken:
+- The `Auth` service registers as a consumer with Kong ([see the Kong documentation](https://docs.konghq.com/2.0.x/getting-started/adding-consumers/#1-create-a-consumer-through-the-restful-api))
+- The `Auth` service requests a new HS256 JWT credential from Kong, used to sign each token
+- All other services, other than `Auth`, are configured through Kong to require a JWT as part of the request body
+
+
+#### Making a Register Request
+Upon making a request to the `/register` endpoint, the service will:
+- Validate the email and password
+  - We currently impose a minimum 8 character limit on the passwords, but you can extend these in your custom defined hooks - see the [Hooks](hooks) guide for more.
+- Validate that a user with that email address doesn't already exist
+- Hash and salt the password using [bcrypt](https://godoc.org/golang.org/x/crypto/bcrypt)
+- Generate a UUID for that user, according to UUID version 1 of [RFC4122](https://tools.ietf.org/html/rfc4122)
+- Generate a HS256 JWT, using the [JWT Go library](https://github.com/dgrijalva/jwt-go), which generates JWTs according to [RFC7519](https://tools.ietf.org/html/rfc7519)
+  - This JWT has 3 claims:
+    1. `id`: the UUID assigned to that entity
+    2. `iss`: the issuer claim (as defined by [RFC7519](https://tools.ietf.org/html/rfc7519))
+    3. `exp`: the expiry of the token, set 24 hours in the future from the current time (again, as defined in [RFC7519](https://tools.ietf.org/html/rfc7519))
```

Comment:
Is this worth explaining? It’s all defined in the RFC, and I’d need to refer to the RFC to understand `iss` anyway

Reply:
I think so, because there are certain claims which we don't include (such as `aud`, `nbf`, `iat` etc.)